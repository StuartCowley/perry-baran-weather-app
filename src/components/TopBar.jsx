import React from "react";
import PropTypes from "prop-types";
import "../styles/TopBar.css";
import SearchForm from "./SearchForm";
import DropDownMenu from "./DropDownMenu";

function TopBar({ handleSearch, location }) {
  return (
    <div className="top-bar">
      <SearchForm handleSearch={handleSearch} placeholder={location} />
      <DropDownMenu handleSearch={handleSearch} />
    </div>
  );
}

TopBar.defaultProps = {
  location: "",
};

TopBar.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  location: PropTypes.string,
};

export default TopBar;
