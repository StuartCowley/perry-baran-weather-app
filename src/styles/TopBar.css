:root {
  --top-bar-height: 60px;
  --top-bar-border: 2px solid var(--dark-blue);
}

.top-bar {
  display: flex;
  justify-content: space-between;
  width: 100vw;
  height: var(--top-bar-height);  
  padding: 10px 10px;
  border-bottom: var(--top-bar-border);
  background-color: var(--medium-blue);
}
