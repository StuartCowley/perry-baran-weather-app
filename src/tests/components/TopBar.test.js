import React from "react";
import { render } from "@testing-library/react";
import TopBar from "../../components/TopBar";

describe("TopBar", () => {
  const handleSearch = jest.fn();

  test("snapshot", () => {
    const { asFragment } = render(<TopBar handleSearch={handleSearch} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
