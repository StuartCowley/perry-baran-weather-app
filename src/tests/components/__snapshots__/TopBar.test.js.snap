// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar snapshot 1`] = `
<DocumentFragment>
  <div
    class="top-bar"
  >
    <div
      class="search-form"
    >
      <input
        placeholder=""
        type="text"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </div>
    <button
      class="drop-down-menu-button"
      type="button"
    >
      <img
        alt="hamburger menu"
        src="menu.png"
      />
    </button>
  </div>
</DocumentFragment>
`;
